src/stdune/string.ml

```diff
@@ -185,6 +185,19 @@ let longest_map l ~f =
 
 let longest l = longest_map l ~f:(fun x -> x)
 
+let longest_prefix = function
```

Comment:
This function is quadratic, it should be linear. I suggest to add a function that returns the longest prefix of two strings and fold it through the list.

Reply:
On the other hand, my implementation does not allocate intermediate strings. Anyways, this is not a critical path so I can change it as you said.

Reply:
What about a function that returns the length of the longest prefix of two strings?

Reply:
It's possible but it will be harder to fold it. What would you expect our folding function to be? I think that we'd have to get current longest prefix as a string to use it with:

```
val longest_prefix : string -> string -> int
```

Which would defeat avoiding the intermediate allocation.

Reply:
If this function is called once only in `dune subst` for jbuilder projects, I don't think that this is worth optimizing it. Just a note about complexity in the `.mli` should be enough.

Reply:
Also, I'm not really sure that the complexity of the current function is strictly worse. For example, it should be preform quite a bit better on lists of strings like:

```
let x = "long string here"
[ x; x; x; ...; "f"]
```

Reply:
My bad, I read too fast and thought `loop` was called inside the `fold_left`